py-rattler/src/lock/mod.rs

```diff
@@ -91,6 +120,40 @@ impl From<PyEnvironment> for Environment {
 
 #[pymethods]
 impl PyEnvironment {
+    #[new]
+    pub fn new(name: String, req: HashMap<PyPlatform, Vec<PyRecord>>) -> PyResult<Self> {
+        let mut lock = LockFile::builder();
+        let channels = req
+            .values()
+            .flat_map(|records| {
+                records
+                    .iter()
+                    .map(record::PyRecord::channel)
+                    .collect::<Vec<PyResult<_>>>()
+            })
+            .collect::<PyResult<HashSet<_>>>()?;
+
+        lock.set_channels(&name, channels);
+
+        for (platform, records) in req {
+            for record in records {
+                lock.add_conda_package(
+                    &name,
+                    platform.inner,
+                    record.try_as_repodata_record()?.clone().into(),
+                );
+            }
+        }
+
+        Ok(lock
+            .finish()
+            .environment(&name)
+            .ok_or(Into::<PyErr>::into(PyRattlerError::RequirementError(
+                "Temp Placeholder Error".to_owned(),
```

Comment:
This seems like something that should be changed ;)

Reply:
Oh no, I forgot this one, thanks alot.